exercises/csharp/TriviaGame/TriviaGame/Game.cs

```diff
@@ -1,211 +1,225 @@
         LinkedList<string> rockQuestions = new LinkedList<string>();
 
-        int currentPlayer = 0;
-        bool isGettingOutOfPenaltyBox;
+        private int currentPlayer;
+        bool isCurrentPlayerLeavingPenaltyBox;
+        int[] playerPlace = new int[maxPlayers];
+        int[] playerPurse = new int[maxPlayers];
+        bool[] inPenaltyBox = new bool[maxPlayers];
+        private int _totalPlaces;
 
         public Game()
         {
-            for (int i = 0; i < 50; i++)
+            for (int i = 0; i < numberOfQuestions; i++)
             {
-                popQuestions.AddLast("Pop Question " + i);
-                scienceQuestions.AddLast(("Science Question " + i));
-                sportsQuestions.AddLast(("Sports Question " + i));
-                rockQuestions.AddLast(CreateRockQuestion(i));
+                popQuestions.AddLast(CreateQuestion(popCategory, i));
+                scienceQuestions.AddLast(CreateQuestion(scienceCategory, i));
+                sportsQuestions.AddLast(CreateQuestion(sportsCategory, i));
+                rockQuestions.AddLast(CreateQuestion(rockCategory, i));
             }
         }
 
-        public string CreateRockQuestion(int index)
+        public string CreateQuestion(string category, int index)
         {
-            return "Rock Question " + index;
+            return $"{category} Question {index}";
         }
 
-        public bool IsPlayable()
+        public bool HasEnoughPlayers()
         {
-            return (HowManyPlayers() >= 2);
+            return (GetTotalPlayers() >= minimumPlayers);
         }
 
-        public bool Add(string playerName)
+        public bool AddPlayer(string playerName)
         {
-
-
-            players.Add(playerName);
-            places[HowManyPlayers()] = 0;
-            purses[HowManyPlayers()] = 0;
-            inPenaltyBox[HowManyPlayers()] = false;
+            playerNames.Add(playerName);
+            inPenaltyBox[GetTotalPlayers()] = false;
 
             Console.WriteLine(playerName + " was added");
-            Console.WriteLine("They are player number " + players.Count);
+            Console.WriteLine("They are player number " + playerNames.Count);
             return true;
         }
 
-        public int HowManyPlayers()
+        public int GetTotalPlayers()
         {
-            return players.Count;
+            return playerNames.Count;
         }
 
-        public void Roll(int roll)
+        public void ProcessPlayerTurn(int rollResult)
         {
-            Console.WriteLine(players[currentPlayer] + " is the current player");
-            Console.WriteLine("They have rolled a " + roll);
-
+            Console.WriteLine(playerNames[currentPlayer] + " is the current player");
+            Console.WriteLine("They have rolled a " + rollResult);
+            
+            _totalPlaces = 12;
             if (inPenaltyBox[currentPlayer])
             {
-                if (roll % 2 != 0)
+                if (isEven(rollResult))
```

Comment:
I'm 99% sure you were right on this being an 'IsOdd' check. 

```suggestion
                if (isOdd(rollResult))
```